src/test/lib/statistics/cardinality_estimator_test.cpp

```diff
@@ -913,4 +914,138 @@ TEST_F(CardinalityEstimatorTest, NonQueryNodes) {
   EXPECT_EQ(estimator.estimate_cardinality(DummyTableNode::make()), 0.0f);
 }
 
+// Usually, we assume that we know nothing about predicates getting their values from uncorrelated subqueries and we
+// simply forward the input estimations. However, special predicates with subqueries can stem from join rewrites (e.g.,
+// JoinToPredicateRewriteRule). We estimate the resulting plans just like we would have done it for the original semi-
```

Comment:
What does the "original semin-join" here refer to? The predicate might come from somewhere else, right?

Reply:
Yes, you're right. In fact, we also improve the estimation queries that were stated exactly like the rewrite. However, the `SubqueryToJoinRule` would have rewritten most of them to (semi-) joins.